cmake/helpers.cmake

```
@@ -28,16 +28,22 @@ function(add_cppformat name)
     if(${ARGC} EQUAL 0)
         return()
     else()
-        add_custom_target(cppformat-${name}
-            COMMAND ${CLANG_FORMAT}
-                --style=file
-                --i
-                ${ARGN}
-            COMMENT "Format CXX source files"
-            )
+        # Split args into 2 parts (in Windows the list is probably too long)
+        list(SUBLIST ARGN 0 250 selected_files)
+        add_custom_target(cppformat-${name}-1
+            COMMAND ${CLANG_FORMAT} --style=file --i ${selected_files}
+            COMMENT "Format CXX source files (pt 1)"
+        )
+
+        list(SUBLIST ARGN 251 -1 selected_files)
+        add_custom_target(cppformat-${name}-2
+            COMMAND ${CLANG_FORMAT} --style=file --i ${selected_files}
+            COMMENT "Format CXX source files (pt 2)"
+        )
```

A custom target can contain multiple commands, there's no need for multiple `cppformat-${name}` targets. e.g.

```cmake
add_custom_target(example
  COMMAND echo first
  COMMAND echo second
  COMMENT Multiple commands example
)
```

Has this output when invoking the `example` target:

```
first
second
```

```suggestion
        # Split args into 2 parts (in Windows the list is probably too long)
        list(SUBLIST ARGN 0 250 selected_files_1)
        list(SUBLIST ARGN 251 -1 selected_files_2)
        add_custom_target(cppformat-${name}
            COMMAND ${CLANG_FORMAT} --style=file --i ${selected_files_1}
            COMMAND ${CLANG_FORMAT} --style=file --i ${selected_files_2}
            COMMENT "Format CXX source files (pt 1)"
        )
```

---

thx! that's super nice, applied.